### PR TITLE
Merge branch 'Dev' of https://github.com/tayhall/Microsoft365DSC into Dev

### PR DESCRIPTION
**Pull request by tayhall**
#### Pull Request (PR) description

* ExternalCrossTenantTrust
  * Added the automatic user consent settings for Inbound and Outbound allowed

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 1]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
         {
             $B2BDirectConnectOutboundValue = $getValue.B2BDirectConnectOutbound
         }
+        if ($null -ne $getValue.AutomaticUserConsentSettings)
+        {
+<<<<<<< HEAD
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:113]: Please make sure to remove the merging comments. These lines should not be visible.

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 2]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
...
+            $AutomaticUserConsentSettingsValue = $getValue.AutomaticUserConsentSettings
+=======
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:115]: see above.

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 3]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
...
+            $AutomaticUserConsentSettings = $getValue.AutomaticUserConsentSettings
+>>>>>>> b2a86a4cd7ddc2a1421f5477b1d4c78bb5a92488
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:117]: see above.

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 4]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
...
-            TenantId                 = $TenantId
-            ApplicationSecret        = $ApplicationSecret
-            CertificateThumbprint    = $CertificateThumbprint
-            ManagedIdentity          = $ManagedIdentity.IsPresent
+            PartnerTenantId              = $getValue.TenantId
+            B2BCollaborationInbound      = $B2BCollaborationInboundValue
+            B2BCollaborationOutbound     = $B2BCollaborationOutboundValue
+            B2BDirectConnectInbound      = $B2BDirectConnectInboundValue
+            B2BDirectConnectOutbound     = $B2BDirectConnectOutboundValue
+<<<<<<< HEAD
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:129]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 5]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
...
+            AutomaticUserConsentSettings = $AutomaticUserConsentSettingsValue
+=======
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:131]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 6]
```diff
@@ -104,24 +108,37 @@ function Get-TargetResource
...
+            AutomaticUserConsentSettings = $AutomaticUserConsentSettings
+>>>>>>> b2a86a4cd7ddc2a1421f5477b1d4c78bb5a92488
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:133]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 7]
```diff
@@ -240,6 +260,14 @@ function Set-TargetResource
         $OperationParams.B2BDirectConnectOutbound = (Get-M365DSCAADCrossTenantAccessPolicyB2BSetting -Setting $OperationParams.B2BDirectConnectOutbound)
         $OperationParams.B2BDirectConnectOutbound = (Update-M365DSCSettingUserIdFromUPN -Setting $OperationParams.B2BDirectConnectOutbound)
     }
+    if ($null -ne $OperationParams.AutomaticUserConsentSettings)
+    {
+<<<<<<< HEAD
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:265]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 8]
```diff
@@ -240,6 +260,14 @@ function Set-TargetResource
...
+        $OperationParams.AutomaticUserConsentSettings = (Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSettings -Setting $OperationParams.AutomaticUserConsentSettings)
+=======
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:267]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 9]
```diff
@@ -240,6 +260,14 @@ function Set-TargetResource
...
+        $OperationParams.AutomaticUserConsentSettings = (Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSetting -Setting $OperationParams.AutomaticUserConsentSettings)
+>>>>>>> b2a86a4cd7ddc2a1421f5477b1d4c78bb5a92488
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:269]: Looks like a duplicate

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 10]
```diff
@@ -477,6 +509,14 @@ function Export-TargetResource
             {
                 $Results.B2BDirectConnectOutbound = Get-M365DSCAADCrossTenantAccessPolicyB2BSettingAsString -Setting $Results.B2BDirectConnectOutbound
             }
+            if ($null -ne $Results.AutomaticUserConsentSettings)
+            {
+<<<<<<< HEAD
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:514]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 11]
```diff
@@ -477,6 +509,14 @@ function Export-TargetResource
...
+                $Results.AutomaticUserConsentSettings = Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSettingsAsString -Setting $Results.AutomaticUserConsentSettings
+=======
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:516]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 12]
```diff
@@ -477,6 +509,14 @@ function Export-TargetResource
...
+                $Results.AutomaticUserConsentSettings = Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSettingAsString -Setting $Results.AutomaticUserConsentSettings
+>>>>>>> b2a86a4cd7ddc2a1421f5477b1d4c78bb5a92488
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:518]: Looks like a duplicate

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 13]
```diff
@@ -702,13 +747,85 @@ function Get-M365DSCAADCrossTenantAccessPolicyB2BSetting
...
     $results = @{
-        Applications = $applications
+        Applications   = $applications
         UsersAndGroups = $usersAndGroups
     }
 
     return $results
 }
 
+<<<<<<< HEAD
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:757]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 14]
```diff
@@ -702,13 +747,85 @@ function Get-M365DSCAADCrossTenantAccessPolicyB2BSetting
...
+function Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSettings
+=======
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:759]: see above

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 15]
```diff
@@ -702,13 +747,85 @@ function Get-M365DSCAADCrossTenantAccessPolicyB2BSetting
...
+function Get-M365DSCAADCrossTenantAccessPolicyAutomaticUserConsentSetting
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:760]: This is a duplicated line

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1 [hunk 16]
```diff
@@ -702,13 +747,85 @@ function Get-M365DSCAADCrossTenantAccessPolicyB2BSetting
...
+>>>>>>> b2a86a4cd7ddc2a1421f5477b1d4c78bb5a92488
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1:761]: see above

**Review by andikrueger — CHANGES_REQUESTED** [review]
There are merging comments within your files. I did not comment on all of them. PLease make sure to remove either the current or incoming change before merging the files.

**Comment by tayhall** [comment]
I don't really follow, have I messed up the first pr and second pr and they
are combined somehow?

**Comment by andikrueger** [comment]
It kind of looks like it. Please make sure to develop your changes within a new branch, that is based on DEV right now, all your changes are made in the dev branch of your fork of the M365DSC repo.

**Comment by tayhall** [comment]
Ok can I just do a fresh pull and start again.

I've copies of my changes that will take 5 minutes to change

**Pull request closed without merging**